src/ui/Base/ComponentOptions.ts

```diff
@@ -687,6 +687,11 @@ export class ComponentOptions {
         template = ComponentOptions.loadResultTemplateFromId(id);
       }
     }
+
+    if (template == null) {
+      template = ComponentOptions.loadResultTemplateFromPageTemplatesCache();
```

Comment:
The `ResultList` doesn't discover the result template to use. It uses its passed "resultTemplate" in the options. In the `resultList` case, the "default" use case really use the `TemplateList` from the `ComponentOptions` to evaluate which template to use.

`resultTemplate: ComponentOptions.buildTemplateOption({ defaultFunction: ResultList.getDefaultTemplate }),`. When no template is passed, it's the only time the `ResultList` is aware of what template to use.

I really followed how and where the `script` tags were loaded and reproduce it with the page templates.